decidim-proposals/spec/controllers/decidim/proposal_endorsements_controller_spec.rb

```diff
@@ -0,0 +1,301 @@
+              get :identities, params: params
+
+              expect(response).to have_http_status(:ok)
+              expect(assigns[:to_endorse_groups]).to be_empty
+              expect(assigns[:to_unendorse_groups]).to be_empty
+            end
+          end
+        end
+
+        context "when endorsements are disabled" do
+          let(:feature) do
+            create(:proposal_feature, :with_endorsements_disabled)
+          end
+
+          it "does not allow endorsing" do
+            expect do
+              post :create, format: :js, params: params
+            end.not_to change { ProposalEndorsement.count }
+
+            expect(flash[:alert]).not_to be_empty
+            expect(response).to have_http_status(302)
+          end
+
+          context "when requesting user identities" do
+            it "raises exception" do
+              get :identities, params: params
+              expect(response).to have_http_status(302)
+            end
+          end
+        end
+
+        context "when endorsements are enabled but endorsements are blocked" do
+          let(:feature) do
+            create(:proposal_feature, :with_endorsements_enabled, :with_endorsements_blocked)
+          end
+
+          it "does not allow endorsing" do
+            expect do
+              post :create, format: :js, params: params
+            end.not_to change { ProposalEndorsement.count }
+
+            expect(flash[:alert]).not_to be_empty
+            expect(response).to have_http_status(302)
+          end
+          context "when requesting user identities" do
+            it "does not allow it" do
+              get :identities, params: params
+              expect(response).to have_http_status(302)
+            end
+          end
+        end
+      end
+      describe "As User unendorsing a Proposal" do
+        before do
+          create(:proposal_endorsement, proposal: proposal, author: user)
+        end
+        context "when endorsements are enabled" do
+          let(:feature) do
+            create(:proposal_feature, :with_endorsements_enabled)
+          end
+
+          it "deletes the endorsement" do
+            expect do
+              delete :destroy, format: :js, params: params
+            end.to change { ProposalEndorsement.count }.by(-1)
+
+            expect(ProposalEndorsement.count).to eq(0)
+          end
+        end
+        context "when endorsements are disabled" do
+          let(:feature) do
+            create(:proposal_feature, :with_endorsements_disabled)
+          end
+
+          it "does not delete the endorsement" do
+            expect do
+              delete :destroy, format: :js, params: params
+            end.not_to change { ProposalEndorsement.count }
+
+            expect(flash[:alert]).not_to be_empty
+            expect(response).to have_http_status(302)
+          end
+        end
+      end
+
+      #
+      # As Organization
+      #
```

Comment:
Redundant comment?

Reply:
Yes, it is redundant, but helps when moving inside the file.
Maybe the file should be split in two, but then all initialization that now is reused will be repeated in both files. Or we can try to use require from inside the spec, I've never tried! should I?

Reply:
You can move the initialization in a `shared_context`, we have some around the code base, so sounds like splitting this into files and using a shared context for the initialization would be awesome 😄

Reply:
:+1: done!